kernel/byterun/coq_uint63_native.h

```diff
@@ -64,17 +64,16 @@
   else \
     accu = uint63_zero(); \
   }while(0)
-/* FIXME this definition is the same as lsr for now */
 #define Uint63_asr(x,y) do{ \
     value uint63_asr_y__ = (y); \
-    if (uint63_asr_y__ < (uint64_t) 127) \
-      accu = (value)(((uint64_t)(x) >> uint63_of_value(uint63_asr_y__)) | 1); \
+    if (uint63_asr_y__ < (int64_t) 127) \
+      accu = (value)(((int64_t)(x) >> uint63_of_value(uint63_asr_y__)) | 1); \
```

Comment:
I just checked the C standard. This code actually implementation-defined for negative `x`. What saves us somehow is that OCaml is currently compiled only on platforms that have the behavior we want for negative `x`. (Either that or there is a bug in OCaml.)